apps/frontend/src/app/5_pages/AavePage/components/LendPositionsList/components/WithdrawModal/WithdrawForm.tsx

```diff
@@ -13,56 +14,75 @@ import {
 import { Decimal } from '@sovryn/utils';
 
+import { BOB_CHAIN_ID } from '../../../../../../../config/chains';
+
 import { AmountRenderer } from '../../../../../../2_molecules/AmountRenderer/AmountRenderer';
 import { AssetAmountInput } from '../../../../../../2_molecules/AssetAmountInput/AssetAmountInput';
 import { AssetRenderer } from '../../../../../../2_molecules/AssetRenderer/AssetRenderer';
+import { useAaveReservesData } from '../../../../../../../hooks/aave/useAaveReservesData';
+import { useAaveUserReservesData } from '../../../../../../../hooks/aave/useAaveUserReservesData';
+import { useAaveWithdraw } from '../../../../../../../hooks/aave/useAaveWithdraw';
 import { useDecimalAmountInput } from '../../../../../../../hooks/useDecimalAmountInput';
 import { translations } from '../../../../../../../locales/i18n';
 
 const pageTranslations = translations.aavePage;
 
 type WithdrawFormProps = {
+  asset: string;
   onSuccess: () => unknown;
 };
 
-export const WithdrawForm: FC<WithdrawFormProps> = () => {
-  const withdrawableAssets = useMemo(() => ['BTC', 'SOV'], []); // TODO: this is mocked data. Replace with proper hook
-  const [maximumWithdrawAmount] = useState<Decimal>(Decimal.from(10)); // TODO: this is mocked data. Replace with proper hook
-  const [withdrawAsset, setWithdrawAsset] = useState<string>(
-    withdrawableAssets[0],
-  );
+export const WithdrawForm: FC<WithdrawFormProps> = ({ asset }) => {
+  const { handleWithdraw } = useAaveWithdraw({});
+  const { reserves } = useAaveReservesData();
+  const { userReservesSummary } = useAaveUserReservesData();
+  const [withdrawAsset, setWithdrawAsset] = useState<string>(asset);
   const [withdrawAmount, setWithdrawAmount, withdrawSize] =
     useDecimalAmountInput('');
 
-  const onWithdrawAssetChange = useCallback(v => {
-    setWithdrawAsset(v);
-  }, []);
-
   const withdrawableAssetsOptions = useMemo(
     () =>
-      withdrawableAssets.map(token => ({
-        value: token,
-        label: (
-          <AssetRenderer
-            showAssetLogo
-            asset={token}
-            assetClassName="font-medium"
-          />
-        ),
-      })),
-    [withdrawableAssets],
+      !userReservesSummary
+        ? []
+        : userReservesSummary.suppliedAssets.map(sa => ({
+            value: sa.asset,
+            label: (
+              <AssetRenderer
+                showAssetLogo
+                asset={sa.asset}
+                assetClassName="font-medium"
+                chainId={BOB_CHAIN_ID}
+              />
+            ),
+          })),
+    [userReservesSummary],
   );
 
-  const isValidWithdrawAmount = useMemo(
-    () => (withdrawSize.gt(0) ? withdrawSize.lte(maximumWithdrawAmount) : true),
-    [withdrawSize, maximumWithdrawAmount],
-  );
+  const maximumWithdrawAmount: Decimal = useMemo(() => {
+    if (!userReservesSummary) return Decimal.from(0);
+    const sa = userReservesSummary.suppliedAssets.find(
+      sa => sa.asset === withdrawAsset,
+    );
```

Comment:
Although for this case it is quite clear thanks of the contexts, I usually avoid using abrev to variables names.

It is too personal anyway but maybe it is useful for you too 💪 

```suggestion
    const suppliedAsset = userReservesSummary.suppliedAssets.find(
      asset => sa.asset === withdrawAsset,
    );
```